course/views.py

```diff
@@ -33,18 +35,16 @@ def get_common_objects(self):
         self.welcome_text = settings_text('WELCOME_TEXT')
         self.internal_user_label = settings_text('INTERNAL_USER_LABEL')
         self.external_user_label = settings_text('EXTERNAL_USER_LABEL')
-        self.instances = []
-        prio2 = []
-        start_treshold = timezone.now() - datetime.timedelta(days=10)
+        self.all_instances = []
+        self.my_instances = []
         end_threshold = timezone.now() - datetime.timedelta(days=30)
         for instance in CourseInstance.objects.get_visible(self.request.user)\
                 .filter(ending_time__gte=end_threshold):
-            if instance.starting_time > start_treshold:
-                self.instances += [instance]
+            if (instance.is_student(self.request.user) or instance.is_course_staff(self.request.user)):
```

Comment:
This looks like this will generate an SQL query per instances, which would be _bad_.  You can show SQL queries in the log with the following settings snippet:

```
LOGGING['loggers'].update({
    'django.db.backends': {
        'level': 'DEBUG',
    },
})
```

I think you need to craft a query which would find the information with a single sql query.

Reply:
Hmm.. This was done in the previous source-code like this way, but I can come up with another way for sure :) ! https://github.com/apluslms/a-plus/blob/master/course/views.py#L41

Reply:
The `start_time` works as that is variable from the list of object returned by the query, but is_student creates [a second query.here](https://github.com/apluslms/a-plus/blob/master/course/models.py#L504). This should be visible with the logging :)

Reply:
Oh you are correct, I wasn't sure which line you were refering to. But yes, will fix that :)

Reply:
Hei, checkout variable `request.user.userprofile.enrolled`. Similar thing is done for the drop down: https://github.com/apluslms/a-plus/blob/master/course/cache/menu.py#L37

Reply:
Did not actually notice this comment, but found the same example myself :D So this has been fixed. For the teaching-part I had to be a little bit more innovative since the relation goes through the course-model. And now it does a few more hits to the database. But only a few more, the code is readable and this affects only teachers who have courses they have teached.